src/gsy_e/models/strategy/scm/smart_meter.py

```diff
@@ -53,9 +53,12 @@ def get_energy_to_buy_kWh(self, time_slot: DateTime) -> float:
     def decrease_energy_to_sell(
             self, traded_energy_kWh: float, time_slot: DateTime, area: "CoefficientArea"):
         """Decrease traded energy from the state and the strategy parameters."""
-        self.state.decrement_energy_requirement(traded_energy_kWh, time_slot, area.name)
+        self.state.decrement_available_energy(traded_energy_kWh, time_slot, area.name)
```

Comment:
The old code was calling the wrong method: when selling energy, the available energy should be decreased.